crates/bevy_ecs/src/system/query.rs

```diff
@@ -931,19 +930,16 @@ where
     /// }
     /// # regenerate_player_health_system.system();
     /// ```
+    #[inline]
     pub fn get_single_mut(
         &mut self,
     ) -> Result<<Q::Fetch as Fetch<'_, '_>>::Item, QuerySingleError> {
-        let mut query = self.iter_mut();
-        let first = query.next();
-        let extra = query.next().is_some();
-
-        match (first, extra) {
-            (Some(r), false) => Ok(r),
-            (None, _) => Err(QuerySingleError::NoEntities(std::any::type_name::<Self>())),
-            (Some(_), _) => Err(QuerySingleError::MultipleEntities(std::any::type_name::<
-                Self,
-            >())),
+        unsafe {
+            self.state.get_single_unchecked_manual(
+                self.world,
+                self.last_change_tick,
+                self.change_tick,
+            )
```

Comment:
I don't think the unsafe variant should be used here.
```suggestion
        self.state.get_single_mut(self.world)
```

Reply:
One big difference is that it uses `self.world.last_change_tick` and `self.wrold.change_tick` instead of `self.last_change_tick ` and `self.change_tick`. I think that's not the same implementation as the original `get_single_mut` unless they are guaranteed to be equal.

Reply:
Hmm. That's a fair point. I am not immediately sure if this is always true.

Reply:
Looking at the implementation of iter_combinations, it does the same procedure of calling the unsafe manual version instead.

As that is the case, I am fine with leaving these two Functions as they currently are.